config/serverless.yml

```diff
@@ -157,6 +157,14 @@ elasticsearch.ignoreVersionMismatch: true
 # Limit maxSockets to 800 as we do in ESS, which improves reliability under high loads.
 elasticsearch.maxSockets: 800
 
+# Hold the DNS lookup resolution for 60s max (if the DNS server claims a shorter TTL, it'll be used instead).
+# This avoids making a dns.lookup request on every ES socket connection.
+# * Improves our startup stability: once the initial connectivity checks are successful,
+#   the subsequent requests will reuse the DNS resolution, making it less prone to suffer from
+#   negative caches during such a critical startup phase.
+# * Hopefully, it may improve performance in sudden bursts of requests, since it doesn't need to re-resolve the IP.
+elasticsearch.dnsCacheTtlInSeconds: 60
```

Comment:
Because this change affects all requests to ES it may make sense to roll this out in a more controlled fashion but I'm not sure what the most effective way of doing that would be. A few approaches:

* Set to `0` (i.e. remove this for now) and turn it on for a set of QA projects and observe them
* Add some logic to our controller to be able set this for a region (probably the most comprehensive test)

Alternatively we can deem this low risk enough and avoid that extra step. And instead monitor the next rollout. WDYT?

---

One observation, based [on this logic](https://github.com/szmarczak/cacheable-lookup/blob/9e60c9f6e74a003692aec68f3ddad93afe613b8f/source/index.mjs#L321) we will be possibly shortening the TTL of ES DNS entries to 60s. I guess that's not a huge problem since our current TTL is 0. Just wanted to point that out!

Reply:
Thanks for the comments! 

> * Set to `0` (i.e. remove this for now) and turn it on for a set of QA projects and observe them

I think this makes a lot of sense. I'd prefer to err on the side of caution and set this in the controllers.
I'll update our default here and work on the controller changes.

> One observation, based [on this logic](https://github.com/szmarczak/cacheable-lookup/blob/9e60c9f6e74a003692aec68f3ddad93afe613b8f/source/index.mjs#L321) we will be possibly shortening the TTL of ES DNS entries to 60s. I guess that's not a huge problem since our current TTL is 0. Just wanted to point that out!

Yes... we won't take advantage of the DNS's server TTL if it's higher than 60s. As you mentioned, today our effective `maxTTL` is 0. So I thought 60s is still a good compromise. FWIW, I didn't want to make this value too large in the event of a topology change (ES changing IPs).